tests/utils/test_validator_state.py

```diff
@@ -0,0 +1,238 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from providers.consensus.typings import ValidatorState
+from src.utils.validator_state import *
+from tests.modules.accounting.bunker.test_bunker_medterm_penalty import simple_validators
+from typings import EpochNumber
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("activation_epoch, epoch, exit_epoch, expected", [
+    (176720, 176720, 176722, True),
+    (176720, 176721, 176722, True),
+    (176900, 176900, 2 ** 64 - 1, True),
+    (176901, 176900, 2 ** 64 - 1, False),
+    (176720, 176720, 176720, False),
+    (176900, 176720, 176720, False),
+    (176900, 176720, 176750, False),
+])
+def test_is_active_validator(activation_epoch, epoch, exit_epoch, expected):
+    validator_state = ValidatorState(
+        activation_epoch=activation_epoch,
+        exit_epoch=exit_epoch,
+        pubkey=MagicMock(),
+        effective_balance=MagicMock(),
+        slashed=MagicMock(),
+        activation_eligibility_epoch=MagicMock(),
+        withdrawable_epoch=MagicMock(),
+        withdrawal_credentials=MagicMock(),
+    )
+
+    validator = Validator(
+        index=MagicMock(),
+        balance=MagicMock(),
+        status=MagicMock(),
+        validator=validator_state,
+    )
+
+    actual = is_active_validator(validator, EpochNumber(epoch))
+    assert actual == expected
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("exit_epoch, epoch, expected", [
+    (176720, 176722, True),
+    (176730, 176722, False),
+    (2 ** 64 - 1, 176722, False),
+])
+def test_is_exited_validator(exit_epoch, epoch, expected):
+    validator_state = ValidatorState(
+        activation_epoch=MagicMock(),
+        exit_epoch=exit_epoch,
+        pubkey=MagicMock(),
+        effective_balance=MagicMock(),
+        slashed=MagicMock(),
+        activation_eligibility_epoch=MagicMock(),
+        withdrawable_epoch=MagicMock(),
+        withdrawal_credentials=MagicMock(),
+    )
+
+    validator = Validator(
+        index=MagicMock(),
+        balance=MagicMock(),
+        status=MagicMock(),
+        validator=validator_state,
+    )
+
+    actual = is_exited_validator(validator, EpochNumber(epoch))
+    assert actual == expected
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize("exit_epoch, expected", [
+    (176720, True),
+    (2 ** 64 - 1, False),
+])
+def test_is_on_exit(exit_epoch, expected):
+    validator_state = ValidatorState(
```

Comment:
please create factory for `ValidatorState`, see `tests/factory` files

Reply:
and reuse some of them

Reply:
Sure!

Reply:
Resolved